containers/src/Data/IntMap/Internal.hs

```diff
@@ -761,6 +764,25 @@ disjoint t1@(Bin p1 m1 l1 r1) t2@(Bin p2 m2 l2 r2)
               | zero p1 m2       = disjoint t1 l2
               | otherwise        = disjoint t1 r2
 
+{--------------------------------------------------------------------
+  Compose
+--------------------------------------------------------------------}
+-- | /O(|ab|*log(|bc|))/. Relate the keys of one map to the values of
```

Comment:
That bound looks wrong. Poke around the rest of the module to see what the bounds look like.

Reply:
Do you mean that it should use `n` & `m` like the other functions? I strayed from that because other bounds seem to be commutative in their arguments, so they don't need to say which argument map is `n` and which is `m`, but `compose` is asymmetric

Reply:
No, I mean that `IntMap` performance doesn't tend to involve logarithms of the size; instead, it usually relies on the word size. This is because the depth of the tree is not bounded by the logarithm of the total size. Check the bounds for `(!?)`, which you use here.

Reply:
Ah yes, makes sense